src/dataland_qa_lab/pages/pages_provider.py

```diff
@@ -0,0 +1,83 @@
+import io
+
+import pypdf
+from dataland_backend.models.nuclear_and_gas_data import NuclearAndGasData
+
+from dataland_qa_lab.dataland.data_provider import DataProvider
+from dataland_qa_lab.utils import config
+
+
+class PagesProvider:
+    """Provide Pages."""
+
+    def __init__(self) -> None:
+        """Initiate class."""
+
+    def get_relevant_pages_of_pdf(self, dataset: NuclearAndGasData) -> pypdf.PdfReader:
+        """Get page numbers of relevant data."""
+        yes_no_pages = self.get_relevant_pages_of_yes_no(dataset=dataset)
+        numeric_pages = self.get_relevant_pages_of_numeric(dataset=dataset)
+        page_numbers = sorted(set(yes_no_pages + numeric_pages))
+
+        file_reference = dataset.general.general.nuclear_energy_related_activities_section426.data_source.file_reference
+
+        dataland_client = config.get_config().dataland_client
+        full_pdf = dataland_client.documents_api.get_document(file_reference)
+        full_pdf_stream = io.BytesIO(full_pdf)
+
+        original_pdf = pypdf.PdfReader(full_pdf_stream)
+        output_pdf = pypdf.PdfWriter()
+
+        for page_num in page_numbers:
+            if 0 <= page_num - 1 < len(original_pdf.pages):
+                output_pdf.add_page(original_pdf.pages[page_num - 1])
+
+        extracted_pdf_stream = io.BytesIO()
+        output_pdf.write(extracted_pdf_stream)
+        extracted_pdf_stream.seek(0)
+
+        return extracted_pdf_stream
+
+    @classmethod
+    def get_relevant_pages_of_yes_no(cls, dataset: NuclearAndGasData) -> list[int]:
```

Comment:
In the future, we might want to extend our software to work on other dataset types as well. Consequently, it might make sense to differentiate between Nuclear-And-Gas specific implementations (like this function), and overarching functionality (extrating pages from PDFs) by e.g., putting them in different classes or at least including the framework-name in the function